config/config_helpers.go

```diff
@@ -318,22 +310,43 @@ var listQuotedParamRegex = regexp.MustCompile(`^"([^"]*?)"\s*,\s*(.*)$`)
 // foo("a", "b") -> return "a", "b", 2, nil
 //
 func parseOptionalQuotedParam(parameters string) (string, string, int, error) {
+	parsed, err := splitParamList(parameters)
```

Comment:
Again, we just extract the full parameter list.

Except here, there were a few tests that called this function with an invalid parameter list.  Now, in the normal flow of code, this would never happen because we're validating that up-front.  But, it's simple enough to validate it again in this function.

Here, I decided to factor out the guts of `parseParamList` to a new function `splitParamList` that would return errors if the input string wasn't a valid parameter-list.  This allows `parseOptionalQuotedParams` to return a proper error if validation fails, as on line 315 below.